File: lib-android/src/main/java/com/pranav/lib_android/util/ConcurrentUtil.java
package com.pranav.lib_android.util;

import java.util.concurrent.Executors;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.TimeUnit;

public class ConcurrentUtil {

  public static ExecutorService createPool() {
    return Executors.newCachedThreadPool();
  }

  public static void execute(Runnable runnable) {
    ExecutorService service = createPool();
    service.execute(runnable);
    service.shutdown();
    try {
      service.awaitTermination(Long.MAX_VALUE, TimeUnit.SECONDS);
    } catch (InterruptedException e) {
      e.printStackTrace();
    }
  }
  
  public static void executeInBackground(Runnable runnable) {
    ExecutorService service = createPool();
    service.execute(runnable);
  }
}

File: lib-android/src/main/java/com/pranav/lib_android/util/ConcurrentUtil.kt
package com.pranav.lib_android.util

import kotlinx.coroutines.*

class ConcurrentUtil {
  companion object {
    @JvmStatic
    fun execute(runnable: Runnable) = runBlocking {
      val deferred: Deferred<Unit> = async {
        runnable.run()
      }
      deferred.await()
    }

    @JvmStatic
    fun executeInBackground(runnable: Runnable) = runBlocking {
      launch {
        runnable.run()
      }
    }
  }
}
